### ISSUE TITLE
Size Class Range Syntax Enhancements

### ISSUE THREAD

**Issue by kohlmannj** [nytpi/nytpi-sass-utils#5]
Currently it's not possible to scale one of the "middle" size classes to a size class larger than itself. Recommend adding a **"for" keyword** to the range syntax, which would let Size Class users specify a named layout to be scaled across multiple size classes. For example:

```scss
$all-tablet: tablet-portrait for mobile-portrait to desktop;
```

This also has the side effect of allowing the range (`n to m`) syntax to be **order-insensitive** when a layout is specified (before `for`):

```scss
// Today, this range would upscale mobile-portrait instead of downscale desktop
$all-desktop-order-insensitive: desktop for mobile-portrait to desktop;
```

Finally, as native media queries already support, adding support for a comma (`,`) token would allow the expression of discontiguous ranges:

```scss
$desktop-for-mobile-landscape-too: desktop for mobile-landscape, tablet-portrait to desktop;
```
----
The addition of the `for` keyword would effectively make the current order-sensitive range syntax a shorthand, which seems like a win-win.

**Comment by kohlmannj** [comment]
The following syntax enhancements have been implemented in the development branch of nytpi-sass-utils in the `size-class-v3` library:

1. Explicit `size-class` argument preceding `for` and an order-indiscriminate range of breakpoints
    - e.g. `desktop for desktop to mobile-portrait`
2. Explicit `layout-type` argument:
    - e.g. `desktop pixel-grid` or `desktop grid`
3. Size families now part of the config and can be referenced by string rather than by variable
    - e.g. `regular`
4. Support for breakpoint ranges in Sass number units
    - e.g. `desktop for 0 to 960px`
5. Support for the `with` keyword to specify size-class and layout-type following a breakpoint range
    - e.g. `tablet-portrait with desktop grid` or `desktop for desktop to tablet-portrait with pixel-grid`
6. breakpoint-end now optional
    - e.g. `desktop for tablet-portrait` (use the `desktop` size class for the `tablet-portrait` breakpoint range) or `desktop for 480px` (use the `desktop` size class within a `min-width: 480px` media query)

**Comment by kohlmannj** [comment]
There's a fix in fb6baab9 that fixes an error with parsing shorthand syntaxes like `desktop pspx` (i.e. `class-name` + `layout-type` without a range of breakpoint names).

**Comment by kohlmannj** [comment]
The above issue is actually *not* fixed—breaking it out into a separate issue in a moment.
